fili-core/src/main/java/com/yahoo/bard/webservice/util/IntervalUtils.java

```diff
@@ -209,4 +211,20 @@ public static long getTotalDuration(Collection<Interval> intervals) {
                 .map(Interval::getStart)
                 .reduce(Utils::getMinValue);
     }
+
+    /**
+     * Returns the coarsest ZonedTimeGrain among a set of ZonedTimeGrains.
+     * <p>
+     * If the set of ZonedTimeGrains is empty, return null.
+     *
+     * @param zonedTimeGrains  A set of ZonedTimeGrains among which the coarsest ZonedTimeGrain is to be returned.
+     *
+     * @return the coarsest ZonedTimeGrain among a set of ZonedTimeGrains
+     */
+    public static ZonedTimeGrain getCoarsestTimeGrain(Collection<ZonedTimeGrain> zonedTimeGrains) {
+        return zonedTimeGrains.stream()
+                .sorted(Comparator.comparing(ZonedTimeGrain::getEstimatedDuration))
```

Comment:
Good point. That's unfortunate. If the intent of this helper method is to make it easy to select the coarsest grain from a collection of PhysicalTables, perhaps we should convert this method to do that, and then use the GranularityComparator?